sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateEncoder.scala

```diff
@@ -29,6 +29,14 @@ sealed trait RocksDBKeyStateEncoder {
   def extractPrefixKey(key: UnsafeRow): UnsafeRow
   def encodeKey(row: UnsafeRow): Array[Byte]
   def decodeKey(keyBytes: Array[Byte]): UnsafeRow
+
+  def encodeCompositeKey(groupingKeyRow: UnsafeRow, userKeyRow: UnsafeRow): Array[Byte] = {
```

Comment:
We can also pass in a two-column key schema to `PrefixKeyScanStateEncoder`.

Reply:
Yes - could we please do this ?